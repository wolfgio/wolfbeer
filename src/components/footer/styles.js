import styled from 'styled-components';

export const FooterContainer = styled.footer`
  border-top: solid 1px #cbcbcb;
  display: flex;
  flex-flow: column wrap;
  align-items: center;
  justify-content: center;
  padding: 16px;
  text-align: center;
`;
