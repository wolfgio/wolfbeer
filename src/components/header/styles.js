import styled from 'styled-components';

export const HeaderContainer = styled.header`
  box-shadow: 1px 4px 5px 1px rgba(0,0,0,0.2);
  display: flex;
  flex-flow: row wrap;
  align-items: center;
  justify-content: flex-start;
  padding: 16px;
`;

export const Icon = styled.img`
  height: 40px;
`;

export const TitleContainer = styled.div`
  display: flex;
  flex-flow: column wrap;
  align-items: flex-end;
  margin-left: 8px;

  h1 {
    font-size: 1.4rem;
  }
`;
