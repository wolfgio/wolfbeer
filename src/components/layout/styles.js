import styled from 'styled-components';

export const Container = styled.div`
  min-height: 100vh;
  display: flex;
  flex: 1;
  flex-flow: column wrap;
`;

export const Content = styled.div`
  display: flex;
  flex-grow: 1;
  flex-basis: 100%;
`;
