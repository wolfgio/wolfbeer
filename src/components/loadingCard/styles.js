import styled from 'styled-components';

export const Container = styled.div`
  height: 280px;
  border-radius: 2px;
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.08);
`;
