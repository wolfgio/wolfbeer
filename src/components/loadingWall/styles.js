import styled, { keyframes } from 'styled-components';

export const WallContainer = styled.div`
  position: absolute;
  z-index: 10;
  background-color: rgba(0, 0, 0, 0.7);
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
`;

const rotate = keyframes`
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
`;

export const SpinnerIcon = styled.img`
  height: 80px;
  animation: ${rotate} 2s linear infinite;
`;
