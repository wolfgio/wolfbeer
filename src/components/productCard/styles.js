import styled from 'styled-components';

export const CardContainer = styled.div`
  border-radius: 2px;
  background-color: #FFF;
  padding: 16px;
  height: 280px;
  position: relative;

  display: flex;
  flex-flow: column wrap;
  justify-content: flex-start;

  box-shadow: 0 1px 3px rgba(0,0,0,0.12), 0 1px 2px rgba(0,0,0,0.24);
  transition: all 0.3s cubic-bezier(.25,.8,.25,1);

  &:hover {
    box-shadow: 0 14px 28px rgba(0,0,0,0.25), 0 10px 10px rgba(0,0,0,0.22);
  }
`;

export const CardTitle = styled.h4`
  font-size: 0.95rem;
  font-weight: 800;
  margin: 16px 0;
  flex-grow: 1;
  display: flex;
  text-align: left;
`;

export const CardImage = styled.img`
  height: 50%;
  object-fit: contain;
`;

export const CardButtonsWrapper = styled.div`
  width: 100%;
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  grid-column-gap: 8px;
`;

export const CardButton = styled.button`
  border: solid 1px ${(props) => props.color};
  border-radius: 20px;
  color: ${(props) => props.color};
  padding: 4px 8px;
  font-size: 0.7rem;
  outline: none;

  &:hover {
    cursor: pointer;
  }
`;
