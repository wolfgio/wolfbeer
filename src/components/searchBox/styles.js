import styled from 'styled-components';

export const BoxContainer = styled.div`
  border-radius: 8px;
  padding: 8px 16px;
  box-shadow: 1px 3px 5px 1px rgba(0,0,0,0.2);
  width: 100%;
  max-width: 680px;
  margin: 0 16px;
  display: flex;
`;

export const Icon = styled.img`
  flex: 0 0 30px;
  height: 30px;
`;

export const SearchInput = styled.input`
  flex: 1 1 100%;
  font-size: 2rem;
  color: #333;
  background-color: transparent;
  border: none;
  outline: none;
  margin-left: 8px;
  font-weight: 300;
  &::placeholder {
    opacity: 0.8;
  }
`;

export const LocationButton = styled.button`
  margin-top: 16px;
  background-color: transparent;
  border: solid 1px transparent;
  border-radius: 8px;
  padding: 8px 16px;
  outline: none;
  vertical-align: middle;
  display: flex;
  align-items: center;
  transition: all .1s;
  opacity: 0.7;

  &:hover {
    cursor: pointer;
    border-color: #333;
  }

  &:active {
    opacity: 0.4;
  }
`;

export const LocationIcon = styled.img`
  height: 16px;
  margin-right: 8px;
`;
