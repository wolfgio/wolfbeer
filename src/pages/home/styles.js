import styled from 'styled-components';

export const Container = styled.div`
  flex-grow: 1;
  flex-basis: 100%;
  display: flex;
  align-items: center;
  flex-flow: column wrap;
  justify-content: center;
`;

export const ErrorContainer = styled.div`
  padding: 16px;
  border-radius: 2px;
  box-shadow: 0 1px 3px rgba(0,0,0,0.12), 0 1px 2px rgba(0,0,0,0.24);
  background-color: #e57373;
  margin-bottom: 16px;
  h4 {
    margin: 0;
    color: #FFF;
  }
`;
