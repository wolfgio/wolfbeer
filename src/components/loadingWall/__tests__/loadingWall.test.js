import React from 'react';
import { create, act } from 'react-test-renderer';

import LoadingWall from '../loadingWall';
import { SpinnerIcon } from '../styles';
import ReloadIcon from '../../../assets/reload.png';

describe('<LoadingCard />', () => {
  it('should render without error', () => {
    let root;
    act(() => {
      root = create(<loadingWall loading />);
    });

    expect(root.toJSON()).toMatchSnapshot();
  });

  it('should render spinner', () => {
    let testRenderer;
    act(() => {
      testRenderer = create(<LoadingWall loading />);
    });

    const testInstance = testRenderer.root;
    expect(testInstance.findByType(SpinnerIcon).props.src).toBe(ReloadIcon);
  });

  it('should not render', () => {
    let testRenderer;
    act(() => {
      testRenderer = create(<LoadingWall loading={false} />);
    });

    const testInstance = testRenderer.root;
    expect(testInstance.children).toStrictEqual([]);
  });
});
