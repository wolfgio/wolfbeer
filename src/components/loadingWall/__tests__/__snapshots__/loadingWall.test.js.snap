// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingWall /> should render without error 1`] = `
<loadingWall
  loading={true}
/>
`;
